server/container/guice/guice-common/src/test/java/org/apache/james/utils/RetryExecutorUtilTest.java

```diff
@@ -0,0 +1,127 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.utils;
+
+import com.nurkiewicz.asyncretry.AsyncRetryExecutor;
+import com.nurkiewicz.asyncretry.RetryExecutor;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.BDDMockito.given;
+
+public class RetryExecutorUtilTest {
+    @Mock
+    protected FaultyService serviceMock;
+
+    private RetryExecutor retryExecutor;
+    private ScheduledExecutorService scheduledExecutor;
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        scheduledExecutor.shutdownNow();
+    }
+
+    @Test
+    public void retryOnClazzAndExecuteShouldRethrowIfOnlyException() throws Exception {
+        given(serviceMock.faultyService()).willThrow(IllegalArgumentException.class);
+        retryExecutor = RetryExecutorUtil.retryOnExceptions(new AsyncRetryExecutor(scheduledExecutor), 2, 3000, IllegalArgumentException.class);
```

Comment:
> Even better it would be great to have a builder for retryOnExceptions

I tried to make it as builder but finally I see it's not acceptable because it's back to the old code with more complicated. 
Basically, the "AsyncRetryExecutor" looks like a builder as it is but we did not used it directly, we would like to build an utility with the testing. The utility is different with builder.
Anyway, I push the changes to my branch without PR: https://github.com/quynhn/james-project/commit/3391ba0cb9ce3a937d58e80d549376463f61456c
Could you take a look and give me your thought?